packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -309,20 +309,68 @@ export class ProfilesUtils {
+            ),
+            {
+                items: [noButton, yesButton],
+                vsCodeOpts: {
+                    modal: true,
+                },
+            }
+        );
+        if (response === yesButton) {
+            await SettingsConfig.setDirectValueForAll("zowe.security.secureCredentialsEnabled", false);
+            await vscode.commands.executeCommand("workbench.action.reloadWindow");
+        } else {
+            throw new imperative.ImperativeError({
+                msg: localize(
+                    "ProfileUtils.promptAndDisableCredentialManagement.error",
+                    // eslint-disable-next-line max-len
+                    "Failed to load credential manager. This may be related to Zowe Explorer being unable to use the default credential manager in a browser based environment."
+                ),
+            });
+        }
+    }
+
     /**
      * Use the default credential manager in Zowe Explorer and setup before use
      * @returns Promise<imperative.ProfileInfo> the object of profileInfo using the default credential manager
      */
     public static async setupDefaultCredentialManager(): Promise<imperative.ProfileInfo> {
-        ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
-        ZoweLogger.info(
-            localize("ProfilesUtils.setupDefaultCredentialManager.usingDefault", "No custom credential managers found, using the default instead.")
-        );
-        await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
-        return new imperative.ProfileInfo("zowe", {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-            credMgrOverride: imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring),
-        });
+        try {
+            ZoweLogger.trace("ProfilesUtils.setupDefaultCredentialManager called.");
+            ZoweLogger.info(
+                localize(
+                    "ProfilesUtils.setupDefaultCredentialManager.usingDefault",
+                    "No custom credential managers found, using the default instead."
+                )
+            );
+            await ProfilesUtils.updateCredentialManagerSetting(globals.ZOWE_CLI_SCM);
+            const defaultCredentialManager = imperative.ProfileCredentials.defaultCredMgrWithKeytar(ProfilesCache.requireKeyring);
+            const profileInfo = new imperative.ProfileInfo("zowe", {
+                credMgrOverride: defaultCredentialManager,
+            });
+            // Trigger initialize() function of credential manager to throw an error early if failed to load
+            await profileInfo.readProfilesFromDisk();
+            return profileInfo;
+        } catch (err) {
+            ZoweLogger.error(err);
+            await ProfilesUtils.promptAndDisableCredentialManagement();
+        }
```

Comment:
This will catch any errors thrown by `readProfilesFromDisk` - file access errors, JSON parse errors, etc., not just credential manager errors.

I think we had a discussion at standup about whether we should enhance Imperative to allow us to catch just the credential manager errors. If that's necessary in order for us to have clear error handling, I'd be happy to help with the Imperative side 😋

Reply:
That would be nice to have, since as you can see in the code we are currently doing a workaround for this since imperative seems to not fail early when calling the constructor but instead fails until the `readProfilesFromDisk` is called